code/modules/cargo/centcom_podlauncher.dm

```diff
@@ -320,71 +415,43 @@
-			if (isnull(timeInput))
-				return
-			if (!isnum(timeInput)) //Sanitize input
-				alert(usr, "That wasn't a number! Value set to default ([initial(temp_pod.openingDelay)*0.1]) instead.")
-				timeInput = initial(temp_pod.openingDelay)
-			temp_pod.openingDelay = 10 *  timeInput
-			. = TRUE
-		if("departureDelay") //Change the time it takes the pod to leave (if bluespace = true it just deletes, if effectReverse = true it goes back to centcom)
-			if (temp_pod.departureDelay != initial(temp_pod.departureDelay)) //If the departure delay has already been changed when we push the "change value" button, then set it to default
-				temp_pod.departureDelay = initial(temp_pod.departureDelay)
-				return
-			var/timeInput = input("Enter the time it takes for the pod to leave after opening, in seconds", "Delay Time", initial(temp_pod.departureDelay) * 0.1) as null|num
-			if (isnull(timeInput))
-				return
-			if (!isnum(timeInput))
-				alert(usr, "That wasn't a number! Value set to default ([initial(temp_pod.departureDelay)*0.1]) instead.")
-				timeInput = initial(temp_pod.departureDelay)
-			temp_pod.departureDelay = 10 * timeInput
+		if("editTiming") //Change the different timers relating to the pod
+			var/delay = params["timer"]
+			var/timer = timers[delay]
+			var/value = params["value"]
+			temp_pod.vars[timer] = value * 10
+			. = TRUE
+		if("resetTiming")
+			for (var/timer in timers)
+				temp_pod.vars[timer] = initial(temp_pod.vars[timer])
 			. = TRUE
-
 		////////////////////////////ADMIN SOUNDS//////////////////
-		if("fallSound") //Admin sound from a local file that plays when the pod lands
+		if("fallingSound") //Admin sound from a local file that plays when the pod lands
 			if ((temp_pod.fallingSound) != initial(temp_pod.fallingSound))
 				temp_pod.fallingSound = initial(temp_pod.fallingSound)
 				temp_pod.fallingSoundLength = initial(temp_pod.fallingSoundLength)
 				return
-			var/soundInput = input(holder, "Please pick a sound file to play when the pod lands! NOTICE: Take a note of exactly how long the sound is.", "Pick a Sound File") as null|sound
+			var/soundInput = input(holder, "Please pick a sound file to play when the pod lands! Sound will start playing and try to end when the pod lands", "Pick a Sound File") as null|sound
 			if (isnull(soundInput))
 				return
-			var/timeInput =  input(holder, "What is the exact length of the sound file, in seconds. This number will be used to line the sound up so that it finishes right as the pod lands!", "Pick a Sound File", 0.3) as null|num
-			if (isnull(timeInput))
-				return
-			if (!isnum(timeInput))
-				alert(usr, "That wasn't a number! Value set to default ([initial(temp_pod.fallingSoundLength)*0.1]) instead.")
+			var/sound/tempSound = sound(soundInput)
+			playsound(holder.mob, tempSound, 1)
+			var/list/sounds_list = holder.SoundQuery()
+			var/soundLen = 0
+			for (var/playing_sound in sounds_list)
+				if (isnull(playing_sound))
+					continue
```

Comment:
I believe this will happen if a sound is right about to end when the soundQuery() proc runs, then by the time we loop over the song it's been deleted. Might happen if there's a large number of songs playing such that looping over them takes enough time.

Reply:
Seems like something that would be unavoidable, but I'll add the stack trace anyways. Also possible that this can't actually happen as I haven't tested this behavior

Reply:
http://www.byond.com/docs/ref/info.html#/client/proc/SoundQuery
Honestly not even totally sure now. The documentation makes it sound like the sound datums returned aren't actually references to the _real_ sounds being played to the client. Any insights on this?